include/util/AppendVariant.h

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) TrajoptLib contributors
+
+#pragma once
+
+#include <variant>
+
+// NOLINTBEGIN
```

Comment:
Why are there NOLINT tags here?

Reply:
`clang-tidy` will get angry about the fact that the parameters aren't used. The parameters are used but only for template type deduction.

Reply:
Add the `[[maybe_unused]]` attribute to the parameters.

Reply:
I've added docs and `[[maybe_unused]]`